### ISSUE TITLE
Problem Porting DQN: Performing one_hot on a tensor

### ISSUE THREAD

**Issue by JadenTravnik** [tensorflow/tfjs-core#550]
I would like to be able to port DQN but am running into an issue where I need to perform a one_hot operation on a tensor.
In tensorflow, I would need to do something like this (python):

     self.targets = tf.placeholder('float32', [None], name='target_q_t')
     self.actions = tf.placeholder('int64', [None], name='action')

     # Gets one_hot of a placeholder
     actions_one_hot = tf.one_hot(self.actions, self.env.action_size, 1.0, 0.0, name='action_one_hot') 

     pred_q = tf.reduce_sum(self.pred_network.outputs * actions_one_hot, reduction_indices=1, name='q_acted')
    self.delta = self.targets - pred_q
    self.clipped_error = tf.where(tf.abs(self.delta) < 1.0,
                                    0.5 * tf.square(self.delta),
                                    tf.abs(self.delta) - 0.5, name='clipped_error')

    self.loss = tf.reduce_mean(self.clipped_error, name='loss')

But in deeplearning.js there doesnt seem to be a way of performing oneHot on a *Tensor*. The only available function is in *NDArrayMath* but as this only accepts *NDArrays*, not *Tensor*s, I am a little confused as to how one might approach this.

**Comment by nsthorat** [comment]
Hi @JadenTravnik, apologies for the confusion.

We stopped developing Graph mode functionality in favor of our new eager mode (which lets you operate on NDArrays and only NDArrays). You can check out some details here: https://github.com/PAIR-code/deeplearnjs/issues/561

We're working on filling out gradients for these ops. If there's something we're missing a gradient for, we can prioritize it.

**Comment by nsthorat** [comment]
We now have `tf.oneHot` which you can use for this in eager mode.
